api/instance/v1/server_utils.go

```diff
@@ -13,6 +13,14 @@ import (
 	"github.com/scaleway/scaleway-sdk-go/utils"
 )
 
+type UpdateServerRequest updateServerRequest
+
+// updateServer update server
+func (s *API) UpdateServer(req *UpdateServerRequest, opts ...scw.RequestOption) (*UpdateServerResponse, error) {
+	defer lockServer(req.Zone, req.ServerID).Unlock()
+	return s.updateServer((*updateServerRequest)(req), opts...)
```

Comment:
Maybe I'm overlooking something, but where do you lock the server resource?

Reply:
in `lockServer` 👍

Reply:
But `lockServer` is executed in the defer?

Reply:
Onyl the last function is deferred

Reply:
Unlock is defered not lock

Reply:
TIL...